bluemira/magnets/strand.py

```diff
@@ -0,0 +1,421 @@
+        percentage:
+            Percentage of each material (with the same ordering of materials).
+        d_strand:
+            Strand diameter in meters.
+        """
+        super().__init__(materials=materials, percentage=percentage, d_strand=d_strand)
+
+    @property
+    def sc_area(self):
+        return self.area * self.percentage[0]
+
+    def Jc(
+            self, B: float, T: float, strain: float = 0.55, T_margin: float = 1.5,
+            **kwargs
+    ) -> float:
+        """
+        Returns the critical current density.
+
+        Parameters
+        ----------
+        B:
+            Operating magnetic field in Teslas.
+        T:
+            Operating temperature in Kelvins.
+        esps:
+            Total applied measured strain in percentage. Default is 0.55.
+        T_margin:
+            Strand temperature margin in operation in Kelvins. Default is 1.5.
+        **kwargs:
+            Additional parameters.
+
+        Returns
+        -------
+        Critical current density in Amperes/m².
+        """
+        return 0
+
+    def Ic(
+            self, B: float, T: float, strain: float = 0.55, T_margin: float = 1.5,
+            **kwargs
+    ) -> float:
+        """
+        Returns the critical current.
+
+        Parameters
+        ----------
+        B:
+            Operating magnetic field in Teslas.
+        T:
+            Operating temperature in Kelvins.
+        esps:
+            Total applied measured strain in percentage. Default is 0.55.
+        T_margin:
+            Strand temperature margin in operation in Kelvins. Default is 1.5.
+        **kwargs:
+            Additional parameters.
+
+        Returns
+        -------
+        Critical current density in Amperes.
+        """
+        return self.Jc(B=B, T=T, strain=strain, T_margin=T_margin) * self.sc_area
+
+
+    def plot_Ic_B(self, B: np.ndarray, T: float, strain: float = 0.55, T_margin: float = 1.5, ax = None, show: bool = True):
+        """
+            Plot the critical current in a range of B
+        """
+        if ax is None:
+            fig, ax = plt.subplots()
+
+        Ic_sc = self.Ic(B=B , T=T, T_margin=T_margin)
+        ax.plot(B, Ic_sc)
+        # Adding the plot title and axis labels
+        plt.title(f"Critical current for {self.__class__.__name__}\n"
+                  f"T = {T} [K], Tmargin = {T_margin} [K], strain = {strain}")  # Title
+        plt.xlabel('B [T]')            # X-axis label
+        plt.ylabel('Ic [A]')           # Y-axis label
+        # Enabling the grid
+        plt.grid(True)
+        if show:
+            plt.show()
+        return ax
+
+class WireNb3Sn(SuperconductingStrand):
+    """Represents an Nb3Sn strand made of 50% Copper100 and 50% Nb3Sn."""
+
+    # superconducting parameters for the calculation of Ic
+    c_ = 1.0
+    Ca1 = 50.06
+    Ca2 = 0.00
+    eps_0a = 0.00312
+    eps_m = -0.00059
+    Bc20max = 33.24
+    Tc0max = 16.34
+    p = 0.593
+    q = 2.156
+    C = 83075 * 1e6
```

Comment:
This is a general parameterisation, that does not represent all Nb3Sn strands. We can separate the data from the parameterisation very easily in this case.